### DISCUSSION TITLE
One Model Per View vs One Model, Similar Views (Headfixed Setups)

### DISCUSSION THREAD

**Discussion by jmdelahanty** [talmolab/sleap#808] [General]
Hey everyone!

I was wondering about whether or not it's possible to use SLEAP for running inference on multiple views if the views are quite similar/the same points are all visible. This is for an application in a headfixed setting for mice.

We're in the process of adapting the work from Nadine Gogolla's lab that can be found [here](https://www.science.org/doi/10.1126/science.aaz9468).

In future recordings we're going to have things oriented either the same way as in their paper for one view or close to it. Here's a schematic from their supplemental figure:

![example_image3](https://user-images.githubusercontent.com/28411193/175117519-d0bf9e4a-0682-4d3b-aaf1-565066d11835.png)

I know that we could train a model for each camera view, but since the data is so similar across views I figured it would be worth trying out whether or not a model could successfully train/do inference for multiple views so long as all the points are visible between them. Intuitively it seems nice to have one model with very similar views because:
- Projects may have camera angles set up slightly differently. It could be somewhat more confusing to train/organize/run multiple models when one slightly more flexible model can be used by everyone
- The model might be more robust in general if given different views (? not sure about this one...)

I'm guessing the advice will be, "Test it out and see what happens!" But just in case this is a bad idea I wanted to reach out and ask.

**Comment by roomrys** [comment]
Hi @jmdelahanty,

To use the same model in both views, you would need to merge the projects together as a [new model is trained](https://github.com/talmolab/sleap/discussions/802#discussioncomment-2995711) from scratch each iteration using the user-labeled frames in the project. 

The views are similar enough that its seems plausible to use labels from one view/project to predict on the new view/merged-project. Definitely worth a shot if you already have ample labeled frames from the old view!

Let us know how it goes!

Thanks,
Liezl

**Reply by jmdelahanty** [reply]
Will keep you posted! We are literally just starting (as you heard in that recent meeting lol) so we have a lot of labeling to do.
